File: LGPL/CommonSoftware/ACSLaser/demo/src/alma/alarmsystemdemo/MountImpl/MountImpl.java
/*******************************************************************************
 * ALMA - Atacama Large Millimeter Array
 * Copyright (c) ESO - European Southern Observatory, 2011
 * (in the framework of the ALMA collaboration).
 * All rights reserved.
 * 
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 * 
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 *******************************************************************************/
package alma.alarmsystemdemo.MountImpl;

import alma.JavaContainerError.wrappers.AcsJContainerServicesEx;
import alma.acs.component.ComponentImplBase;
import alma.alarmsystemdemo.MountOperations;
import alma.alarmsystemdemo.Antenna;
import alma.alarmsystemdemo.AntennaOperations;
import alma.alarmsystemdemo.AntennaHelper;

import alma.alarmsystem.source.ACSAlarmSystemInterfaceFactory;
import alma.alarmsystem.source.ACSFaultState;
import alma.alarmsystem.source.ACSAlarmSystemInterface;
import java.sql.Timestamp;
import java.util.Properties;

class MountImpl extends ComponentImplBase implements MountOperations {
	Antenna antenna;
	public void faultMount() {
		m_containerServices.getAlarmSource().raiseAlarm("Mount","ALARM_SOURCE_MOUNT",1);
		try { 
			Thread.sleep(5000);
		} catch (Exception e) {}
		Antenna ant = getAntenna();
		if (ant!=null) {
			ant.faultAntenna();
		}
	}
	public void terminate_faultMount() {
		m_containerServices.getAlarmSource().clearAlarm("Mount","ALARM_SOURCE_MOUNT",1);
		try { 
			Thread.sleep(5000);
		} catch (Exception e) {}
		Antenna ant = getAntenna();
		if (ant!=null) {
			ant.terminate_faultAntenna();
		}
	}
	
	private Antenna getAntenna() {
		if (this.antenna==null) {
			org.omg.CORBA.Object cmp = null;
			try
			{
				cmp = m_containerServices.getComponent("ALARM_SOURCE_ANTENNA");
				antenna = AntennaHelper.narrow(cmp);
			}
			catch (Exception ex)
			{}
		}
		return antenna;
	}
}

File: Benchmark/componentsLoading/test/MountImpl.cpp
/*******************************************************************************
*    ALMA - Atacama Large Millimiter Array
*    (c) European Southern Observatory, 2002
*    Copyright by ESO (in the framework of the ALMA collaboration)
*    and Cosylab 2002, All rights reserved
*
*    This library is free software; you can redistribute it and/or
*    modify it under the terms of the GNU Lesser General Public
*    License as published by the Free Software Foundation; either
*    version 2.1 of the License, or (at your option) any later version.
*
*    This library is distributed in the hope that it will be useful,
*    but WITHOUT ANY WARRANTY; without even the implied warranty of
*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
*    Lesser General Public License for more details.
*
*    You should have received a copy of the GNU Lesser General Public
*    License along with this library; if not, write to the Free Software
*    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
* 
*/

#include <MountImpl.h>

using namespace acscomponent;
using namespace testManager;

using std::string;
using std::auto_ptr;

MountImpl::MountImpl(const ACE_CString &name,maci::ContainerServices * containerServices) :
    ACSComponentImpl(name, containerServices)
{
	ACS_TRACE("::MountImpl::MountImpl");

}

MountImpl::~MountImpl()
{
	ACS_TRACE("::MountImpl::~MountImpl");
}

void MountImpl::objfix(CORBA::Double az, CORBA::Double el)
{
	AUTO_TRACE("MountImpl::objfix");
	ACS_SHORT_LOG((LM_INFO,"Pointing to [%f,%f]",az,el));
}

/**
 *  Life cycle method
 */
void MountImpl::initialize() {
	AUTO_TRACE("MountImpl::initialize");
	sleep(1);
}

/**
 *  Life cycle method
 */
void MountImpl::execute() {
	AUTO_TRACE("MountImpl::execute");
}

/**
 *  Life cycle method
 */
void MountImpl::cleanUp() {
	AUTO_TRACE("MountImpl::cleanUp");
}

 /**
  *  Life cycle method
  */
void MountImpl::aboutToAbort() {
	AUTO_TRACE("MountImpl::aboutToAbort");
}

/* --------------- [ MACI DLL support functions ] -----------------*/

#include <maciACSComponentDefines.h>
MACI_DLL_SUPPORT_FUNCTIONS(MountImpl)
/* ----------------------------------------------------------------*/
